### PR TITLE
ruff: manual docstring rules

### PR DESCRIPTION
**Pull request by ahms5**
- add the checks for the docsstring, see ``pyproject.toml``
- full list can be found here: https://docs.astral.sh/ruff/rules/#pydocstyle-d

I have added this rules because accoring to my understanding they should be included anyway. 
We could also think about adding 
D100 - D107 to make sure that every public class, function, module, etc has a documentation. for now we dont have it, we could also exclude the tests in this case
- [x] add D415, D417

update since we follow numpy docstyle we can choose the numpy setting in ruff see pr #685 :
https://docs.astral.sh/ruff/faq/#does-ruff-support-numpy-or-google-style-docstrings

### CODE REVIEW

**Review by f-brinkmann — DISMISSED** [review]
I'm fine with the changes caused by this and I would also support adding D100 - D106. In our case D107 does not seem so useful because we usually document the __init__ as part of the class Docstring.

**Comment by ahms5** [comment]
Thanks for your feedback. I've created a new pr for D100-D107 in #614

File: pyproject.toml [hunk 1]
```diff
@@ -5,10 +5,24 @@ exclude = [
     "setup.py",
 ]
 line-length = 79
-lint.ignore = []
+lint.ignore = [
```
Comment (pingelit) [inline pyproject.toml:8]: Maybe remove this change?

**Review by pingelit — APPROVED** [review]
Thanks, this look very nice. Just one very minor comment.

File: pyproject.toml [hunk 2]
```diff
@@ -5,10 +5,24 @@ exclude = [
...
+]
```
Comment (ahms5) [inline pyproject.toml:9]:
```suggestion
lint.ignore = []
```

File: pyproject.toml (hunk 1 above)
Comment (ahms5) [reply pyproject.toml]: done

**Review by pingelit — APPROVED** [review]

**Review by f-brinkmann — DISMISSED** [review]
There are 3 ruff errors left, that seem to be easy to fix. Approved otherwise - thanks for the effort

**Comment by ahms5** [comment]
merging dev into this pr required more fixes, please review again @f-brinkmann and others.

File: pyfar/classes/audio.py [hunk 3]
```diff
@@ -373,6 +373,7 @@ def time(self, value):
 
     @property
     def complex(self):
+        """Return or set the flag indicating if the data is complex."""
```
Comment (f-brinkmann) [inline pyfar/classes/audio.py:376]: Maybe more explicit: `"""Return or set the flag indicating if the time data is complex."""`. The property does not exist for Frequency data.

File: pyfar/classes/audio.py [hunk 4]
```diff
@@ -965,6 +966,7 @@ def sampling_rate(self, value):
...
```
Comment (f-brinkmann) [inline pyfar/classes/audio.py:969]: See comment above.

**Review by f-brinkmann — APPROVED** [review]
Thanks a lot - this looks like it was quite a bunch of work. I only found one added docstring that might be slightly improved.

**Review by f-brinkmann — APPROVED** [review]

File: pyfar/io/_codec.py [hunk 5]
```diff
@@ -340,11 +340,12 @@ def _encode_ndarray(ndarray):
...
     ----------
-    ndarray: numpy.array.
+    ndarray: numpy.array
+        The numpy array that should be encoded.
 
     Returns
     -------
-    bytes.
+    bytes : bytes
         They bytes that where written by `numpy.save` into a memfile.
```
Comment (hoyer-a) [inline pyfar/io/_codec.py:349]:
```suggestion
        The bytes that were written by `numpy.save` into a memfile.
```

File: pyfar/plot/_interaction.py [hunk 6]
```diff
@@ -570,7 +570,7 @@ def toggle_side(self):
 
     @property
     def plot_type(self):
-        """Return current the plot type"""
+        """Return current the plot type."""
```
Comment (hoyer-a) [inline pyfar/plot/_interaction.py:573]:
```suggestion
        """Return the current plot type."""
```

**Review by hoyer-a — COMMENTED** [review]
Just two small suggestions. Thanks for the effort!

**Review by hoyer-a — APPROVED** [review]

**Comment by f-brinkmann** [comment]
Yes, I think so

**Pull request merged**